classes/section/Collector.php

```diff
@@ -166,7 +166,7 @@ public function getQueryBuilder(): Builder
                     $qb->select('p.*')
                         ->from('publications AS p')
                         ->whereNotNull('p.' . $this->dao->primaryKeyColumn)
-                        ->where(DB::raw('p.' . $this->dao->primaryKeyColumn, '=', 's.' . $this->dao->primaryKeyColumn))
```

Comment:
Was this actually not working in the previous form?

Reply:
The sample below outputs `p.x`:
```php
$primaryKeyColumn = 'x';
echo DB::raw('p.' . $primaryKeyColumn, '=', 's.' . $primaryKeyColumn); 
```

Reply:
Just wondering whether an important scenario wasn't tested. This is [only used in OMP](https://github.com/pkp/omp/blob/main/pages/catalog/CatalogHandler.php#L102), and it would be complicated or impossible to get the wrong data to show up due to this. Cherry-picked into `main`, thanks!